src/main/java/AudioPlayer.java

```diff
@@ -0,0 +1,240 @@
+     *
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public void shuffle() throws IOException, InterruptedException {
+        if (validateSongs(songs)) {
+            Random randomSongIndex = new Random();
+            do {
+                currentSongIndex = randomSongIndex.nextInt(songs.size());
+                System.out.println("Currently playing :" + getSongInfo());
+            } while (!executeSong());
+        }
+    }
+
+    /**
+     * This method search the list of songs by the song's title and return the singer's name and song number in the list.
+     *
+     * @param title - this parameter is the title of the song to be searched by
+     * @return String - this method return the singer of the song and it's number in the list.
+     */
+    public String searchByTitle(String title) {
+        if (title != null && !title.isEmpty()) {
+            for (Song song : songs) {
+                if (song.getTitle().equalsIgnoreCase(title)) {
+                    StringBuilder songInfo = new StringBuilder("Singer: ");
+                    songInfo.append(song.getSinger().getName())
+                            .append("\nSong's number: ")
+                            .append(songs.indexOf(song) + 1);
+                    return songInfo.toString();
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * This method return list of songs in string format searched by the singer of the song.
+     *
+     * @param singer - this parameter is the singer of the song, whom we are looking for
+     * @return String - this method return list of songs in string format
+     */
+    public String searchBySinger(Singer singer) {
+        StringBuilder songsBySinger = new StringBuilder();
+        songsBySinger.append("Songs by: ")
+                .append(singer.getName());
+
+        for (Song song : songs) {
+            if (song.getSinger().equals(singer)) {
+                songsBySinger.append(song);
+            }
+        }
+        return songsBySinger.toString();
+    }
+
+    /**
+     * This method return number of songs that are present in the list
+     *
+     * @return int - the returned value is the count of the songs in the list
+     */
+    public int size() {
+        return songs.size();
+    }
+
+    /**
+     * This method adds a new song to the list
+     *
+     * @param song - this parameter is the actual song
+     */
+    public void add(Song song) {
+        if (song != null) {
+            songs.add(song);
+        }
+    }
+
+    /**
+     * This method first check if the song is present in the list and then deletes it from the list
+     *
+     * @param song - this parameter is the song to be deleted
+     */
+    public void remove(Song song) {
+        if (songs.contains(song)) {
+            songs.remove(song);
+            currentSongIndex--;
```

Comment:
This change of `currentIndex` is not correct! If we have 5 songs and decide to remove the last one, but the current index is 2... find the answer :)